src/main/java/com/project/controller/ApplicationController.java

```diff
@@ -24,7 +24,7 @@ public ApplicationController(ApplicationService applicationService) {
     }
 
     @GetMapping
-    public ResponseEntity<List<Application>> getAll() {
+    public ResponseEntity<List<Application>> fetchAllApplications() {
```

Comment:
You don't need to explicitly say that it will fetch `all` applications. It is already clear from you `Get` since it has no request parameters.